diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 86bace7..2081bf8 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -12,6 +12,10 @@ def safeAppExtGet(prop, fallback) {
     appProject?.ext?.has(prop) ? appProject.ext.get(prop) : fallback
 }
 
+def getNodeExecutable() {
+    return safeAppExtGet("NODE_BINARY", System.getenv("NODE_BINARY") ?: "node")
+}
+
 def isNewArchitectureEnabled() {
     // To opt-in for the New Architecture, you can either:
     // - Set `newArchEnabled` to true inside the `gradle.properties` file
@@ -30,7 +34,7 @@ def resolveReactNativeDirectory() {
     def reactNativePackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native/package.json')")
+            commandLine(getNodeExecutable(), "--print", "require.resolve('react-native/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativePackage.exists()) {
@@ -52,7 +56,7 @@ def resolveReactNativeWorkletsDirectory() {
     def reactNativeWorkletsPackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native-worklets/package.json')")
+            commandLine(getNodeExecutable(), "--print", "require.resolve('react-native-worklets/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativeWorkletsPackage.exists()) {
@@ -268,7 +272,7 @@ android {
 
 def validateReactNativeVersionResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
+    commandLine(getNodeExecutable(), "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
     ignoreExitValue = true
 }
 
@@ -293,7 +297,7 @@ preBuild.dependsOn(assertNewArchitectureEnabledTask)
 
 def validateWorkletsBuildResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-worklets-build.js")
+    commandLine(getNodeExecutable(), "./../scripts/validate-worklets-build.js")
     ignoreExitValue = true
 }
 
